newapi/ooniapi/probe_services.py

```diff
@@ -142,6 +165,17 @@ def check_in():
     run_type = data.get("run_type", "timed")
     charging = data.get("charging", True)
 
+    resp = dict(
+        v=1
+    )
+
+    if probe_cc == "ZZ" and asn == "AS0":
+        probe_ip = get_probe_ip()
+        probe_cc = lookup_probe_cc(probe_ip)
```

Comment:
We should handle exceptions here probably. Also what do we want to do for unknown ipaddrs/cc?

Reply:
> Also what do we want to do for unknown ipaddrs/cc?

Yeah this is a good point. I guess that for those cases we should, as usual, default to `ZZ` and `AS0` as we usually do, but we might want to also additionally either log this somehow or return the fact that the lookup call failed to the caller via the API response.

Logging is a bit dangerous and tricky, because we don't want to be logging IP addresses of users, but at the same time we would collect some amount of information that will allow us to investigate why the lookup failed so that we can improve our databases.

What are your thoughts?

> Also, should we build a database that returns both CC and ASN in one lookup?

That would be ideal, however we currently have in place only code for generating the ASN database ourselves and we don't have a good strategy for building an IP to CC database since we use the one from https://db-ip.com/.

I think we can start off with this approach and document switching to a single database for future work.

Reply:
As an initial iteration we can log the lookup failure and generate a metric without logging the ipaddr.

In future, if the license allows it and if the benefit is worth the effort, we can preprocess the db-ip.com database for faster lookup.

Can you handle lookups leading to unknown / None in this PR?